### ISSUE TITLE
keras_word2vec.py doesn't converge

### ISSUE THREAD

**Issue by romeokienzler** [adventuresinML/adventures-in-ml-code#7]
Hi, please see below - 
fyi - running on python 2.7

Iteration 75600, loss=0.764099955559
Iteration 75700, loss=0.703801393509
Iteration 75800, loss=0.786265671253
Iteration 75900, loss=0.683647096157
Iteration 76000, loss=0.696249544621
Iteration 76100, loss=0.686785459518
Iteration 76200, loss=0.641143143177
Iteration 76300, loss=0.66484028101
Iteration 76400, loss=0.673577725887
Iteration 76500, loss=0.706019639969
Iteration 76600, loss=0.662336111069
Iteration 76700, loss=0.693303346634
Iteration 76800, loss=0.667164921761
Iteration 76900, loss=0.699192345142
Iteration 77000, loss=0.666543602943
Iteration 77100, loss=0.750823795795
Iteration 77200, loss=0.216024711728
Iteration 77300, loss=0.774852216244
Iteration 77400, loss=0.725894510746
Iteration 77500, loss=0.800480008125
Iteration 77600, loss=0.599587082863
Iteration 77700, loss=0.712833166122
Iteration 77800, loss=0.707559227943
Iteration 77900, loss=0.757015168667
Iteration 78000, loss=0.791208267212
Iteration 78100, loss=0.773834228516
Iteration 78200, loss=0.702738165855
Iteration 78300, loss=0.701055109501
Iteration 78400, loss=0.707786381245
Iteration 78500, loss=0.702587544918
Iteration 78600, loss=1.01350021362
Iteration 78700, loss=0.684418618679
Iteration 78800, loss=0.670415282249
Iteration 78900, loss=0.692475199699
Iteration 79000, loss=0.699831724167
Iteration 79100, loss=0.66406583786
Iteration 79200, loss=0.572940707207
Iteration 79300, loss=0.687783002853
Iteration 79400, loss=0.693389892578
Iteration 79500, loss=0.518278539181
Iteration 79600, loss=0.726815402508
Iteration 79700, loss=0.696648955345
Iteration 79800, loss=0.739838123322
Iteration 79900, loss=0.70836687088
Iteration 80000, loss=0.712565600872
Nearest to however: idle, paperback, boris, consolidated, preserved, protest, africans, pointing,
Nearest to four: bremen, vi, fire, designations, citing, ruth, flash, flanders,
Nearest to such: mathrm, adaptive, urban, places, radio, exhibit, corporate, meets,
Nearest to world: shelter, elite, jet, protons, evident, somalia, original, democrats,
Nearest to were: clan, expectancy, comprises, compiler, persians, maxwell, defining, allah,
Nearest to eight: involves, d, hearts, bit, apart, player, press, orthodox,
Nearest to that: prevention, warrior, include, treaty, congo, belief, aerospace, dia,
Nearest to can: waterways, gwh, chord, marriages, rituals, crossing, defended, known,
Nearest to while: dial, exhibits, selective, leonard, extensions, concern, perfectly, egyptian,
Nearest to or: eventually, heard, organised, mirrors, piano, blessed, touch, crowd,
Nearest to after: lab, track, eritrea, implemented, fl, papacy, history, mpeg,
Nearest to first: cp, obvious, demons, allowing, libya, watching, prototype, mistake,
Nearest to use: corner, greenwich, neighbours, biology, converted, armenia, superhero, welsh,

**Comment by kevkid** [comment]
I am experiencing the same thing.

**Comment by yiyang-yu** [comment]
Hi, thanks for the nice tutorial! While running the tutorial `keras_word2vec` (on python 3.5, keras 2.2.2, TensorFlow 1.10.1), I observe similar issue: the loss does not converge, however the model seems to ''learn'' something.

```
Nearest to d: two, seven, three, eight, nine, s, one, at,
Nearest to eight: two, six, seven, nine, a, s, of, the,
Nearest to one: of, that, is, this, was, not, with, seven,
Nearest to most: this, in, not, for, at, an, their, its,
Nearest to as: and, an, is, this, of, that, has, for,
Nearest to used: the, is, s, of, and, with, in, it,
Nearest to world: some, a, and, had, in, all, it, the,
Nearest to first: s, two, in, some, his, on, a, this,
Nearest to they: in, not, that, is, as, some, of, and,
Nearest to if: their, was, has, as, for, united, five, the,
Nearest to was: is, has, that, of, this, be, an, not,
Nearest to by: also, the, has, as, was, of, some, are,
Nearest to but: be, that, was, it, this, the, has, is,
Nearest to b: one, four, three, had, seven, nine, to, two,
Nearest to for: has, as, is, in, that, and, to, s,
Nearest to their: has, is, for, was, the, that, by, this,
Iteration 190100, loss=0.706157922744751
Iteration 190200, loss=0.8479189872741699
Iteration 190300, loss=0.20626628398895264
Iteration 190400, loss=0.03240996226668358
Iteration 190500, loss=0.2608884572982788
Iteration 190600, loss=1.1691124439239502
Iteration 190700, loss=0.5178102254867554
Iteration 190800, loss=0.3880608379840851
Iteration 190900, loss=0.8529599905014038
Iteration 191000, loss=0.6620449423789978
Iteration 191100, loss=0.9693207144737244
Iteration 191200, loss=0.3847825825214386
Iteration 191300, loss=0.4094489514827728
Iteration 191400, loss=0.3334934711456299
Iteration 191500, loss=0.49866509437561035
Iteration 191600, loss=0.02301456406712532
Iteration 191700, loss=0.2802729308605194
Iteration 191800, loss=0.363527774810791
Iteration 191900, loss=0.03329170495271683
Iteration 192000, loss=0.13125231862068176
Iteration 192100, loss=0.003010867163538933
Iteration 192200, loss=1.3108644485473633
Iteration 192300, loss=2.5082199573516846
Iteration 192400, loss=0.00940511655062437
Iteration 192500, loss=0.2341649830341339
Iteration 192600, loss=0.4802365303039551
Iteration 192700, loss=1.5100440979003906
Iteration 192800, loss=0.50730299949646
Iteration 192900, loss=0.2910715937614441
Iteration 193000, loss=0.29807090759277344
Iteration 193100, loss=0.748119592666626
Iteration 193200, loss=0.7718933820724487
Iteration 193300, loss=1.652421474456787
Iteration 193400, loss=0.04813004285097122
Iteration 193500, loss=0.09783909469842911
Iteration 193600, loss=0.7674630284309387
Iteration 193700, loss=0.415813684463501
Iteration 193800, loss=0.269938200712204
Iteration 193900, loss=0.05480138584971428
Iteration 194000, loss=2.352586507797241
Iteration 194100, loss=0.4252009093761444
Iteration 194200, loss=1.3005433082580566
Iteration 194300, loss=0.572454571723938
Iteration 194400, loss=0.34925997257232666
Iteration 194500, loss=0.91258704662323
Iteration 194600, loss=0.2614150047302246
Iteration 194700, loss=0.4022839367389679
Iteration 194800, loss=0.48827385902404785
Iteration 194900, loss=0.02030140347778797
Iteration 195000, loss=0.20791220664978027
Iteration 195100, loss=0.3829302489757538
Iteration 195200, loss=0.39047229290008545
Iteration 195300, loss=0.1228068545460701
Iteration 195400, loss=0.42020970582962036
Iteration 195500, loss=0.5152478814125061
Iteration 195600, loss=1.7521439790725708
Iteration 195700, loss=1.2493896484375
Iteration 195800, loss=0.3689422905445099
Iteration 195900, loss=1.3971350193023682
Iteration 196000, loss=0.6385608315467834
Iteration 196100, loss=0.6653118133544922
Iteration 196200, loss=0.3625309467315674
Iteration 196300, loss=0.072109654545784
Iteration 196400, loss=0.6722363233566284
Iteration 196500, loss=0.15822277963161469
Iteration 196600, loss=0.0025786003097891808
Iteration 196700, loss=0.31820234656333923
Iteration 196800, loss=0.35205912590026855
Iteration 196900, loss=0.4894079864025116
Iteration 197000, loss=0.08752074837684631
Iteration 197100, loss=0.22296302020549774
Iteration 197200, loss=0.4964912533760071
Iteration 197300, loss=0.6326013207435608
Iteration 197400, loss=0.9483761787414551
Iteration 197500, loss=0.2605926990509033
Iteration 197600, loss=0.291573703289032
Iteration 197700, loss=0.35525965690612793
Iteration 197800, loss=0.5912957191467285
Iteration 197900, loss=0.8584250211715698
Iteration 198000, loss=0.022425489500164986
Iteration 198100, loss=0.1733468472957611
Iteration 198200, loss=0.235866978764534
Iteration 198300, loss=0.857978105545044
Iteration 198400, loss=0.2226078063249588
Iteration 198500, loss=0.5773928165435791
Iteration 198600, loss=0.09592520445585251
Iteration 198700, loss=0.37856245040893555
Iteration 198800, loss=0.21361462771892548
Iteration 198900, loss=0.3806821405887604
Iteration 199000, loss=0.3437630534172058
Iteration 199100, loss=0.5689221620559692
Iteration 199200, loss=0.3436322808265686
Iteration 199300, loss=1.1832385063171387
Iteration 199400, loss=0.2717357873916626
Iteration 199500, loss=0.20231099426746368
Iteration 199600, loss=0.8902583122253418
Iteration 199700, loss=0.8637596368789673
Iteration 199800, loss=0.8615012168884277
Iteration 199900, loss=0.28978657722473145
```
I am trying to understand this. Could you please confirm it is still working on your side?
Thanks!
